File: day-7/part-2.py
from statistics import mean
from math import factorial

# https://adventofcode.com/2021/day/7

with open('input.txt') as infile:
	positions = [int(i) for i in infile.readline().split(',')]
	bestCost = None
	for target in range(min(positions), max(positions)):
		cost = sum([((abs(i - target) - 1) * (abs(i - target) / 2) + abs(i - target)) for i in positions])
		if bestCost is None or cost < bestCost:
			bestCost = cost
			
	print(bestCost)

File: day-19/part-2.ts
import { load, Coord, scannerOffsets } from './part-1';

// https://adventofcode.com/2021/day/19

function manhattanDistance(c1: Coord, c2: Coord) {
    return Math.abs(c2[2] - c1[2]) + Math.abs(c2[1] - c1[1]) + Math.abs(c2[0] - c1[0])
}

let bestDistance = 0;
for (const c1 of scannerOffsets) {
    for (const c2 of scannerOffsets) {
        bestDistance = Math.max(bestDistance, manhattanDistance(c1, c2))
    }
}

console.log('part 2:', bestDistance);
